### PR TITLE
CATTY-400 Implement StoreProjectReporter

### PR DESCRIPTION
**Pull request by aeluna**
Added StoreProjectReporter. Created Extension for ProjectDetailStoreViewController, in order to use StoreProjectReporter for reporting a Project. Added DVR tests for StoreProjectReporter. 

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by roelac11 — APPROVED** [review]

**Review by m-herold — CHANGES_REQUESTED** [review]
Thank you very much! Please rebase your branch and remove the two `NSLog` statements from the `ProjectDetailStoreViewController` extension.

File: src/Catty/Store/StoreProjectReporter.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        self.session = session
+    }
+
+    func report(projectId: String, message: String, completion: @escaping (StoreProjectReporterError?) -> Void) {
+
+        let urlString = NetworkDefines.reportProjectUrl
+        var request = URLRequest.init(url: URL(string: urlString)!)
+
+        guard let postData = String(format: "program=%@&note=%@&category=%@", projectId, message, standardCategory).data(using: .utf8, allowLossyConversion: true) else { return }
+        let contentType = "multipart/form-data; boundary=\(httpBoundary)"
```
Comment (m-herold) [inline src/Catty/Store/StoreProjectReporter.swift:48]: Please remove `contentType`. This is a normal POST request and not a _multipart/form-data_ request.

File: src/Catty/Store/StoreProjectReporter.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        if let token = JNKeychain.loadValue(forKey: NetworkDefines.kUserLoginToken) as? String {
+            let tokenString = String(format: "Bearer %@", token)
```
Comment (m-herold) [inline src/Catty/Store/StoreProjectReporter.swift:51]: Please remove the "Bearer " prefix and send the token alone.

**Review by m-herold — CHANGES_REQUESTED** [review]
The API changed on the API side, so that your request now theoretically works. However, there are a few changes required which I left directly at the relevant lines of code. Also, please display the message "Your report was successfully sent!" when the report was successfully sent (instead of "Reported project"). Thanks!

File: src/Catty/Store/StoreProjectReporter.swift (hunk 1 above)
Comment (aeluna) [reply src/Catty/Store/StoreProjectReporter.swift]: Removed!

File: src/Catty/Store/StoreProjectReporter.swift (hunk 2 above)
Comment (aeluna) [reply src/Catty/Store/StoreProjectReporter.swift]: Removed!

**Review by m-herold — APPROVED** [review]

**Pull request merged**